tools/dds/dds-adapter/lrs-device-controller.cpp

```diff
@@ -464,6 +456,28 @@ find_profile( std::shared_ptr< realdds::dds_stream_server > const & stream,
 }
 
 
+std::shared_ptr< realdds::dds_stream_server >
+lrs_device_controller::frame_to_streaming_server( rs2::frame const & f, rs2::stream_profile * p_profile ) const
```

Comment:
Why do we need `p_profile` and all this code handling it? Just to avoid calling `f.get_profile()` outside this function?

Reply:
Well, yes -- it's actually two calls into librealsense and populations of data. Why do something twice?

Reply:
Much less readable, not relevant for all cases and I am not sure if it more efficient...
Had to read several times to understand all this handling with profile/p_profile/profile_.
Passing another parameter, creating a default initialized profile that is not always needed, `if` statement... just to avoid an API call...